src/lib/mathlib/math/Functions.hpp

```diff
@@ -166,28 +166,47 @@ const T interpolate(const T &value, const T &x_low, const T &x_high, const T &y_
 }
 
 /*
- * Constant, linear, linear, constant function with the three corner points as parameters
- *  y_high               -------
- *                      /
- *                    /
- *  y_middle        /
- *                /
+ * Constant, piecewise linear, constant function with 1/N size intervalls and N corner points as parameters
+ * y[N]                 -------
+ *                     /
+ *                   /
+ * y[1]            /
  *               /
  *              /
- * y_low -------
- *         x_low x_middle x_high
+ *             /
+ * y[0] -------
+ *        0 1/(N-1) 2/(N-1) ... 1
  */
-template<typename T>
-const T interpolate3(const T &value,
-		     const T &x_low, const T &x_middle, const T &x_high,
-		     const T &y_low, const T &y_middle, const T &y_high)
+template<typename T, size_t N>
+const T interpolateN(const T &value, const T(&y)[N])
 {
-	if (value < x_middle) {
-		return interpolate(value, x_low, x_middle, y_low, y_middle);
+	size_t index = constrain((int)(value * (N - 1)), 0, (int)(N - 2));
+	return interpolate(value, (T)index / (T)(N - 1), (T)(index + 1) / (T)(N - 1), y[index], y[index + 1]);
+}
 
-	} else {
-		return interpolate(value, x_middle, x_high, y_middle, y_high);
+/*
+ * Constant, piecewise linear, constant function with N corner points as parameters
+ * y[N]                 -------
+ *                     /
+ *                   /
+ * y[1]            /
+ *               /
+ *              /
+ *             /
+ * y[0] -------
+ *          x[0] x[1] ... x[N]
```

Comment:
Should be named : y[N-1] and x[N-1]

Reply:
You're right in my example above it's 5 Anchor points (N = 5)
x[0], x[1], x[2], x[3], x[4]
Let me correct that.

Reply:
Fixed in a43f207